test/zzz-rsyslog.sh

```diff
@@ -224,3 +229,82 @@ os::cmd::expect_success_and_not_text "cat $ops | jq -r .hits.hits[0]._source.lev
 os::cmd::expect_success_and_not_text "cat $ops | jq -r .hits.hits[0]._source.hostname" "^null$"
 ts=$( cat $ops | jq -r '.hits.hits[0]._source."@timestamp"' )
 os::cmd::expect_success "test ${ts} != null"
+
+# Test logrotation (LOG400) - OKD 4.2 and above
+# check rsyslog logs
+rpod=$( get_running_pod rsyslog )
+oc logs $rpod -c rsyslog > $ARTIFACT_DIR/rsyslog.log 2>&1
+oc exec $rpod -c rsyslog -- ls -l /var/log/rsyslog/rsyslog.log > $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+oc exec $rpod -c rsyslog -- cat /var/log/rsyslog/rsyslog.log >> $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+os::cmd::expect_success_and_text "oc logs $rpod -c rsyslog | grep 'oc exec <pod_name> -- logs'" "^oc exec .pod_name. -- logs$"
+os::cmd::expect_success "oc exec $rpod -c rsyslog -- ls /var/log/rsyslog/rsyslog.log > /dev/null 2>&1"
+logsize=$( oc exec $rpod -c rsyslog -- wc -c /var/log/rsyslog/rsyslog.log | awk '{print $1}' )
+
+# Check if logrotate works as expected
+if [ $logsize -gt 0 ]; then
+    # set max log file count and max log size
+    maxcount=3
+    maxsize=$( expr $logsize / $maxcount )
+    oc set env daemonset/rsyslog LOGGING_FILE_SIZE=$maxsize LOGGING_FILE_AGE=$maxcount
+
+    # run logrotate every minute for testing
+    savecm=$( mktemp )
+    workcm=$( mktemp )
+    oc get configmap logrotate-crontab -o yaml > $savecm
+    cat $savecm | sed -e 's,\([ ]*\)[0-9]* .* \(root[ 	]*/usr/bin/bash[ 	]*/opt/app-root/bin/logrotate.*.sh\),\1* * * * *       \2,' > $workcm
+    cp $savecm $ARTIFACT_DIR/logrotate-crontab.orig.yaml
+    cp $workcm $ARTIFACT_DIR/logrotate-crontab.yaml
+    if [ -s $workcm ]; then
+        oc apply --force -f $workcm
+    else
+        artifact_log WARNING generated logrotate-crontab is empty.
+    fi
+    restart_rsyslog_pod $rpod
+    # wait longer than ($maxcount + 1) * 60 seconds.
+    sleep $( expr $( expr $maxcount + 1 ) \* 60 )
+
+    rpod=$( get_running_pod rsyslog )
+    filecount=$( oc exec $rpod -c rsyslog -- ls -l /var/log/rsyslog/ | grep rsyslog.log- | wc -l )
+    filesize=$( oc exec $rpod -c rsyslog -- ls -l /var/log/rsyslog/ | grep "rsyslog.log$" | awk '{print $5}' )
+    artifact_log "logrotate crontab"
+    kubectl exec $rpod -c logrotate -it /usr/bin/cat /etc/cron.d/logrotate 2>&1 | artifact_out
```

Comment:
why use `kubectl` instead of `oc`?

Reply:
I don't know the reason, but oc exec / rsh -c logrotate does not work for me, just kubectl exec...

Reply:
ok - that's for `rsh` for an interactive session - but `oc exec` does not need an interactive session like `rsh`.  I'm also not sure why you would need `-it` for `exec`.   You only need to pass `-i` if you expect the exec'd command to read from stdin and you only need `-t` if stdin is a terminal.  I don't think you are doing either one here.  Does
```
     oc exec $rpod -c logrotate -- /usr/bin/cat /etc/cron.d/logrotate 2>&1 | artifact_out
```
work?

Reply:
Yes!  (Sorry, somehow I was thinking in the different way... :p)  I'm updating the test.

$ oc exec rsyslog-s9znq -c logrotate -- /usr/bin/cat /etc/cron.d/logrotate
# crontab
# login as root
# run logrotate (/var/log/rsyslog/*.log) at 2am everyday
# run logrotate_pod (/var/lib/rsyslog.pod) at 3am everyday
0 2 * * *       root /usr/bin/bash /opt/app-root/bin/logrotate.sh
0 3 * * *       root /usr/bin/bash /opt/app-root/bin/logrotate_pod.sh